Rubberduck.CodeAnalysis/Inspections/Abstract/ImplicitWorkbookReferenceInspectionBase.cs

```diff
@@ -12,33 +12,50 @@ internal ImplicitWorkbookReferenceInspectionBase(IDeclarationFinderProvider decl
             : base(declarationFinderProvider)
         { }
 
-        private static readonly string[] InterestingMembers =
+        protected virtual string[] InterestingMembers => new[]
         {
             "Worksheets", "Sheets", "Names"
         };
 
-        private static readonly string[] InterestingClasses =
+        protected virtual string[] InterestingClasses => new[]
         {
-            "_Global", "_Application", "Global", "Application"
+            "_Global", "_Application", "Global", "Application", "_Workbook", "Workbook"
         };
 
+        private IReadOnlyList<ModuleDeclaration> _relevantClasses;
+        private IReadOnlyList<PropertyGetDeclaration> _relevantProperties;
+
+        protected Declaration Excel { get; private set; }
+
         protected override IEnumerable<Declaration> ObjectionableDeclarations(DeclarationFinder finder)
         {
-            if (!finder.TryFindProjectDeclaration("Excel", out var excel))
+            if (Excel == null)
```

Comment:
Unfortunately, I have not seen this before the PR got merged. This caching is problematic for two reasons.

1. This is not thread-safe although there are multiple implementations of this base class that will get executed in parallel. Usually, one could use a `Lazy<T>`, but this is not possible here since the value depends on a parameter, the instance of `DeclarationFinder`. This immediately brings me to the other problem.
2. This breaks the inspection in case the user unloads and then reloads the Excel library, e.g. in an Access project that creates Excel reports. (Imagine switching between compilation parameters back and forth while developing to verify that everything also works late-bound.)

I think, although the cashing might be good for performance, we really should not do it here.